### PR TITLE
Script for bernoulli naive bayes algorithm added

### PR DESCRIPTION
**Pull request by AayushK47**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by AayushK47** [comment]
@cozek @cclauss can someone please review this pull request? Please!!

File: machine_learning/bernoulli_naive_bayes.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+for text classification and recommender systems.
+
+You can find the detailed notes on this algorithm contributed by me on this notebook: -
+    https://git.io/JTkvO
+"""
+import numpy as np
+import pandas as pd
+
+
+class naive_bayes:
```
Comment (cclauss) [inline machine_learning/bernoulli_naive_bayes.py:18]:
```suggestion
class NaiveBayes:
```
Python naming...  See CONTRIBUTING.md

File: machine_learning/bernoulli_naive_bayes.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    def __init__(self, X, y):
```
Comment (cclauss) [inline machine_learning/bernoulli_naive_bayes.py:19]: Python type hints?  See CONTRIBUTING.md

Single-letter names for function parameters look so old-school and do not help the caller.

File: machine_learning/bernoulli_naive_bayes.py [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+        # Initializing the column name y. (came in handy for me.
+        # If you do not require it, then you can delete it)
+        self.y_label = y.columns[0]
+
+        # Initializing the variables to store class priors. Initiallt set to None.
+        # They will be the assigned the correct values by executing calculate_prior
+        # p_pos is probability of positive class
+        # p_neg is probability of negative class
+        self.p_pos = None
```
Comment (cclauss) [inline machine_learning/bernoulli_naive_bayes.py:39]:
```suggestion
        self.probability_pos = None
```
Don't force the reader study to comments.

File: machine_learning/bernoulli_naive_bayes.py [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+                    0
+                    if feature_given_no == 0
+                    else np.log(feature_given_no / total_negative)
+                )
+
+                # Store the likelihood the the dict
+                self.likelihoods[feature_name][f"{feature}|yes"] = feature_given_yes
+                self.likelihoods[feature_name][f"{feature}|no"] = feature_given_no
+
+    def predict(self, test_data):
```
Comment (cclauss) [inline machine_learning/bernoulli_naive_bayes.py:127]: Type hints?

File: machine_learning/bernoulli_naive_bayes.py [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+                "Yes",
+                "No",
+            ],
+        }
+    )
+
+    X = df.iloc[:, :-1]
+    y = df.iloc[:, -1:]
+
+    model = naive_bayes(X, y)
```
Comment (cclauss) [inline machine_learning/bernoulli_naive_bayes.py:275]:
```suggestion
    model = NaiveBayes(X, y)
```

**Review by cclauss — CHANGES_REQUESTED** [review]
What is `X` and what is `y` and what does the prediction do to make the world a better place?  We need to tell a story so the reader cares.  Are we saving patients or tracking stocks or doing automated driving?  What is the real-world reason for doing this?  Without that, this feels too much like a how-to example.

**Pull request closed without merging**